Remove the embedder InlinePredictionsEnabled preference from UnifiedWebPreferences

rr-codes opened the pull request:
#### c82ca3ff5ea27d224713169862dd81e323391dd4
<pre>
Remove the embedder InlinePredictionsEnabled preference from UnifiedWebPreferences
<a href="https://bugs.webkit.org/show_bug.cgi?id=259236">https://bugs.webkit.org/show_bug.cgi?id=259236</a>
rdar://112304979

Reviewed by Tim Horton.

Since this preference isn&apos;t read in WebCore, or in the WebProcess, we can simply read off the
WKWebViewConfiguration, without a need for a preference associated with the API.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _initializeWithConfiguration:]):
(-[WKWebView _setupPageConfiguration:]):
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::setInlinePredictionsEnabled):
(WebKit::WebViewImpl::inlinePredictionsEnabled const):
(WebKit::WebViewImpl::allowsInlinePredictions const):

Canonical link: <a href="https://commits.webkit.org/266279@main">https://commits.webkit.org/266279@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/359b728538608f106170e59f590b36cd6e41bd13

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/13389 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/13703 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/14035 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/15126 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/12744 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/16210 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/13729 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/15425 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/13556 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/14208 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/11328 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/15798 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/11496 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/12083 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/19137 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/11410 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/12571 "Passed tests") | [💥 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/12250 "An unexpected error occured. Recent messages:OS: Monterey (12.6.7), Xcode: 14.0; Skipping applying patch since patch_id isn't provided; Checked out pull request") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/15468 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/12674 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/12753 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/10611 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/13434 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/12024 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/3522 "Passed tests") | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3268 "Failed to push commit to Webkit repository") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/16347 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/13818 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/12595 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/3319 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebKit/UIProcess/mac/WebViewImpl.h:
```diff
@@ -696,6 +696,11 @@ ALLOW_DEPRECATED_DECLARATIONS_END
     void cancelTextRecognitionForVideoInElementFullscreen();
 
+#if HAVE(INLINE_PREDICTIONS)
+    void setInlinePredictionsEnabled(bool);
```
Not entirely clear why these are out of line (but, also fine...)

rr-codes replied:
good point. will fix

rr-codes committed:
Remove the embedder InlinePredictionsEnabled preference from UnifiedWebPreferences
https://bugs.webkit.org/show_bug.cgi?id=259236
rdar://112304979

Reviewed by Tim Horton.

Since this preference isn't read in WebCore, or in the WebProcess, we can simply read off the
WKWebViewConfiguration, without a need for a preference associated with the API.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _initializeWithConfiguration:]):
(-[WKWebView _setupPageConfiguration:]):
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::setInlinePredictionsEnabled):
(WebKit::WebViewImpl::inlinePredictionsEnabled const):
(WebKit::WebViewImpl::allowsInlinePredictions const):

Canonical link: https://commits.webkit.org/266279@main
